### ISSUE TITLE
`no-indented-block` (`E112`) rule is not needed because it's already covered by `syntax-error` (`E999`)

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#6931]
```py
for item in items:
pass
```
```
main.py:2:1: E112 Expected an indented block
main.py:2:1: E999 SyntaxError: Expected 'Indent', but got 'pass'
```

since `E112` is a nursery rule i think it should just be removed, especially since there are false positives (eg. #6930)

**Comment by charliermarsh** [comment]
I think I'm okay with this because, while rare, it can _also_ catch cases that _aren't_ syntax errors, like:

```python
def print_list(my_list):
# Some comment
    print('Nope')
```

**Comment by DetachHead** [comment]
i know black covers that case, looks like `ruff format` does too https://play.ruff.rs/36beb564-71e4-4e1e-a402-e67569688a97?secondary=Format
